Add Agentic Memory APIs (#10714)
opensearch-project/documentation-website: _ml-commons-plugin/api/agentic-memory-apis/get-memory.md

---
layout: default
title: Get Agentic Memory
parent: Agentic Memory APIs
grand_parent: ML Commons APIs
nav_order: 10
---

# Get Agentic Memory API
**Introduced 3.2**
{: .label .label-purple }

Use this API to retrieve a memory by its ID.

## Endpoint

```json
GET /_plugins/_ml/memory_containers/{memory_container_id}/memories/{memory_id}
```

## Example request

```json
GET /_plugins/_ml/memory_containers/SdjmmpgBOh0h20Y9kWuN/memories/T9jtmpgBOh0h20Y91WtZ
```
{% include copy-curl.html %}

## Example response

```json
{
    "session_id": "sess_a99c5a19-cee3-44ce-b64d-6fbdc411c537",
    "memory": "Machine learning is a subset of artificial intelligence",

>     "memory_type": "RAW_MESSAGE",

Reviewer: Try to mention that currently we only have `RAW_MESSAGE` and `FACT` types

    "user_id": "admin",

>     "role": "assistant",

Reviewer 2: What are all possible roles?

Author: Roles can be `assistant` or `user`

Reviewer: Actually we don't have a constraint on that. It's just in current LLMs `user` and `assistant` are most common.

    "memory_type": "RAW_MESSAGE",
    "user_id": "admin",
    "role": "assistant",
    "created_time": 1754945934681,
    "last_updated_time": 1754945934681
}
```

## Response body fields

The following table lists all response body fields.

| Field               | Data type | Description                                                                              |
| :------------------ | :-------- | :--------------------------------------------------------------------------------------- |
| `session_id`        | String    | The unique identifier for the session associated with this memory.                       |
| `memory`            | String    | If the memory was created with `infer: false`, contains the stored text from the message. If the memory was created with `infer: true`, contains the extracted fact from the message.                                       |
| `memory_type`       | String    | The type of memory. `RAW_MESSAGE` indicates the unprocessed message text. `FACT` indicates a fact inferred by the large language model. |
| `user_id`           | String    | The ID of the user associated with this memory.                                          |

> | `role`              | String    | The role of the message author, such as `assistant` and `user`.                       |

Reviewer 2: ```suggestion
| `role`              | String    | The role of the message author (`assistant` or `user`).                       |
```

> | `role`              | String    | The role of the message author. Can be any string, such as`assistant` or `user`.                      |

Reviewer 3: ```suggestion
| `role`              | String    | The role of the message author. Can be any string, such as `assistant` or `user`.                      |
```
